sixtyfps_compiler/passes.rs

```diff
@@ -105,14 +105,14 @@ pub async fn run_passes(
         lower_popups::lower_popups(component, &doc.local_registry, diag);
         lower_layout::lower_layouts(component, &mut type_loader, diag).await;
         z_order::reorder_by_z_order(component, diag);
-        lower_shadows::lower_shadow_properties(component, &doc.local_registry, diag);
-        clip::handle_clip(component, &global_type_registry.borrow(), diag);
         transform_and_opacity::handle_transform_and_opacity(
             component,
             &global_type_registry.borrow(),
             diag,
         );
         default_geometry::default_geometry(component, diag);
```

Comment:
I hope that the default_geometry is not needed to apply the geometry to these elements.

Reply:
Good question. I checked this now:

 * The clip pass explicitly installs bindings for the `width` and `height` properties of the newly created `Clip` element (that sits between the `Rectangle` and its children)
 * The shadows pass installs bindings for `x`/`y`/`width` and `height` on the newly created `BoxShadow` sibling that refer to the original element.
 * The situation where the `BoxShadow` becomes the new root of a repeated element moves the bindings to the geometry properties from the "old root` to the newly created `BoxShadow`. However the previous root does not have bindings for `width` and `height` anymore and relies on the default geometry pass. There's also a comment that underlines this:

>//  Finally we default geometry pass following this shadow lowering will apply a binding to
>//  the width and height of the inner to follow the size of the parent (the shadow).

This means that this patch would break the default geometry for Rectangles in repeaters that use a box-shadow and this patch needs fixing (code and comment).